functions-and-operators/json-functions/json-functions-modify.md

```diff
@@ -0,0 +1,327 @@
+
+An alias to [`JSON_ARRAY_APPEND()`](#json_array_append).
+
+## [JSON_ARRAY_APPEND()](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-array-append)
+
+The `JSON_ARRAY_APPEND(json_array, path, val)` function appends values to the end of the indicated arrays within a JSON document at the specified `path` and returns the result.
+
+This function takes arguments in pairs, where each pair is a `path` and a `value`.
+
+Examples:
+
+The example below adds an item to an array, which is the root of the JSON document.
+
+```sql
+SELECT JSON_ARRAY_APPEND('["Car", "Boat", "Train"]', '$', "Airplane") AS "Transport options";
+```
+
+```
++--------------------------------------+
+| Transport options                    |
++--------------------------------------+
+| ["Car", "Boat", "Train", "Airplane"] |
++--------------------------------------+
+1 row in set (0.00 sec)
+```
+
+The example below adds an item to an array at the specified path.
+
+```sql
+SELECT JSON_ARRAY_APPEND('{"transport_options": ["Car", "Boat", "Train"]}', '$.transport_options', "Airplane") AS "Transport options";
+```
+
+```
++-------------------------------------------------------------+
+| Transport options                                           |
++-------------------------------------------------------------+
+| {"transport_options": ["Car", "Boat", "Train", "Airplane"]} |
++-------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+## [JSON_ARRAY_INSERT()](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-array-insert)
+
+The `JSON_ARRAY_INSERT(json_array, path, str)` function inserts a value into the specified locations of a JSON array and returns the result.
+
+This function takes arguments in pairs, where each pair is a `path` and a `value`.
+
+Examples:
+
+In this example we add a value at position 0 of the array.
+
+```sql
+SELECT JSON_ARRAY_INSERT('["Car", "Boat", "Train"]', '$[0]', "Airplane") AS "Transport options";
+```
+
+```
++--------------------------------------+
+| Transport options                    |
++--------------------------------------+
+| ["Airplane", "Car", "Boat", "Train"] |
++--------------------------------------+
+1 row in set (0.01 sec)
+```
+
+In this example we add a value at position 1 of the array.
+
+```sql
+SELECT JSON_ARRAY_INSERT('["Car", "Boat", "Train"]', '$[1]', "Airplane") AS "Transport options";
+```
+
+```
++--------------------------------------+
+| Transport options                    |
++--------------------------------------+
+| ["Car", "Airplane", "Boat", "Train"] |
++--------------------------------------+
+1 row in set (0.00 sec)
+```
+
+## [JSON_INSERT()](https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#function_json-insert)
+
+The `JSON_INSERT(json_doc, path, value [,path, value] ...)` function inserts one or more values into a JSON document and returns the result.
```

Comment:
Should we mention that it will not overwrite existing paths?

Reply:
I'll add an example to clarify that.